src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -418,12 +418,14 @@ Optional<TaskNotificationType> updateReplicateStatusWithoutThreadSafety(String c
             replicate.setChainCallbackData(updateReplicateStatusArgs.getChainCallbackData());
         }
 
-        if (statusUpdate.getDetails() != null && statusUpdate.getDetails().getStdout() != null) {
-            if (statusUpdate.getStatus().equals(COMPUTED)) {
-                String stdout = statusUpdate.getDetails().tailStdout().getStdout();
-                stdoutService.addReplicateStdout(chainTaskId, walletAddress, stdout);
-            }
-            statusUpdate.getDetails().setStdout(null);
+        if (statusUpdate.getDetails() != null && statusUpdate.getDetails().getStdout() != null &&
+                (COMPUTED.equals(statusUpdate.getStatus())
+                || (COMPUTE_FAILED.equals(statusUpdate.getStatus())
+                        && ReplicateStatusCause.APP_COMPUTE_FAILED.equals(statusUpdate.getDetails().getCause())))){
```

Comment:
Can `statusUpdate.getDetails()` be `null` there?

Reply:
I see your point, updated thanks :)

Reply:
Actually it can be null, but I updated/refactored check here https://github.com/iExecBlockchainComputing/iexec-core/blob/feature/task-feedback/src/main/java/com/iexec/core/replicate/ReplicatesService.java#L428